src/ctxmenu.ts

```diff
@@ -178,11 +187,21 @@ class ContextMenu implements CTXMenuSingleton {
     }
 
     private static addStylesToDom() {
+        // Only add styles once
+        if (this.stylesAdded) return;
+        
         if (document.readyState === "loading") {
             return document.addEventListener("readystatechange", this.addStylesToDom, { once: true });
         }
+        
+        this.stylesAdded = true;
+        
         //insert default styles as first css -> low priority -> user can overwrite it easily
         const style = document.createElement("style");
+        const nonce = ContextMenu.instance?.nonce;
```

Comment:
I'd make `nonce` a static member, this way you can access it with `this.nonce` here.